src/include/sof/spinlock.h

```diff
@@ -84,7 +85,7 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 #define spinlock_init(lock) \
 	do { \
 		arch_spinlock_init(lock); \
-		(lock)->user = __LINE__; \
+		(*lock)->user = __LINE__; \
```

Comment:
Lovely macros again. If I see correctly `lock` is pointer or pointer to pointer, so we can easily turn this macro into static inline and get type-checking.

Reply:
Should I do it in this PR (separate commit) or leave it as an issue so it can be discussed more?

Reply:
Approving this one, but let's follow up with what @jajanusz suggested.

Reply:
Feel free to do it in separeate commit/PR